sumologic/async.py

```diff
@@ -101,12 +109,14 @@ def __init__(self, accessId, accessKey, endpoint=None, cookieFile='cookies.txt')
 
     async def sources(self, collector_id, limit=None, offset=None):
         params = {'limit': limit, 'offset': offset}
-        r = await self.get('/collectors/' + str(collector_id) + '/sources', params)
-        return json.loads(await r.text())['sources']
+        async with r:
+            r = await self.get('/collectors/' + str(collector_id) + '/sources', params)
+            return json.loads(await r.text())['sources']
 
     async def source(self, collector_id, source_id):
-        r = await self.get('/collectors/' + str(collector_id) + '/sources/' + str(source_id))
-        return json.loads(await r.text()), r.headers['etag']
+        async with r:
+            r = await self.get('/collectors/' + str(collector_id) + '/sources/' + str(source_id))
+            return json.loads(await r.text()), r.headers['etag']
 
     async def create_source(self, collector_id, source):
         return await self.post('/collectors/' + str(collector_id) + '/sources', source)
```

Comment:
i cannot prevent leaking here without changing the API, as this returns the request object itself.